dashboard/pages/primary_screening/callbacks.py

```diff
@@ -0,0 +1,106 @@
+from dash import html, no_update, callback, Output, Input, State
+
+from ..error import throwable
+from ...data.bmg_plate import parse_bmg_files_from_iostring
+from ...data.file_preprocessing.echo_files_parser import EchoFilesParser
+import io
+import base64
+import functools
+import uuid
+
+
+@throwable(1)
+def upload_bmg_data(contents, names, last_modified, stored_uuid, file_storage):
+    if contents is None:
+        return no_update
+
+    if not stored_uuid:
+        stored_uuid = str(uuid.uuid4())
+
+    bmg_files = []
+
+    for content, filename in zip(contents, names):
+        name, extension = filename.split(".")
+        if extension == "txt":
+            _, content_string = content.split(",")
+            decoded = base64.b64decode(content_string)
+            bmg_files.append((filename, io.StringIO(decoded.decode("utf-8"))))
+
+    if len(bmg_files):
```

Comment:
you do not have to use `if len(*any_kind_of_list)` to check if list is empty, just use `if *any_kind_of_list*` without calling `len`